server/gateway/src/odspUtils.ts

```diff
@@ -36,11 +41,73 @@ export function isSpoServer(server: string) {
     return false;
 }
 
+// TODO: These functions are taken from @fluidframework/odsp-utils package and should be removed
+// once the updated getDriveItemByRootFileName function in PR #3826 is available on public feeds
+function toIODSPDriveItem(parsedDriveItemBody: any): IOdspDriveItem {
+    const path = parsedDriveItemBody.parentReference.path !== undefined ?
+        parsedDriveItemBody.parentReference.path.split("root:")[1] : "/";
+    return {
+        path,
+        name: parsedDriveItemBody.name,
+        drive: parsedDriveItemBody.parentReference.driveId,
+        item: parsedDriveItemBody.id,
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        isFolder: !!parsedDriveItemBody.folder,
+    };
+}
+
+async function getDriveItem(
+    getDriveItemUrl: string,
+    authRequestInfo: IOdspAuthRequestInfo,
+    create: boolean,
+): Promise<IOdspDriveItem> {
+    let getDriveItemResult = await getAsync(getDriveItemUrl, authRequestInfo);
+    if (getDriveItemResult.status !== 200) {
+        if (!create) {
+            throw createErrorFromResponse("Unable to get drive/item id from path", getDriveItemResult);
+        }
+
+        // Try creating the file
+        const contentUri = `${getDriveItemUrl}/content`;
+        const createResult = await putAsync(contentUri, authRequestInfo);
+        if (createResult.status !== 201) {
+            throw createErrorFromResponse("Failed to create file.", createResult);
+        }
+
+        getDriveItemResult = await getAsync(getDriveItemUrl, authRequestInfo);
+        if (getDriveItemResult.status !== 200) {
+            throw createErrorFromResponse("Unable to get drive/item id from path", getDriveItemResult);
```

Comment:
Should we make this response separate from the above. Maybe, `unable to get drive/item id from path after create` or something

Reply:
Same here, I just took this from odsp-utils so if we want to make changes, we should probabyl do it directly there